diff --git a/launcher.c b/launcher.c
index 727f7916..431e0339 100644
--- a/launcher.c
+++ b/launcher.c
@@ -35,8 +35,6 @@
 
 #pragma comment (lib, "Shlwapi.lib")
 
-#define APPENDED_ARCHIVE
-#define USE_ENVIRONMENT
 #define SUPPORT_RELATIVE_PATH
 
 #define MSGSIZE 1024
@@ -822,6 +820,13 @@ run_child(wchar_t * cmdline)
 #endif
         si.dwFlags |= STARTF_USESTDHANDLES;
     }
+
+    size_t rez_envvar_size = 0;
+    getenv_s(&rez_envvar_size, NULL, 0, "REZ_LAUNCHER_DEBUG");
+    if (rez_envvar_size > 0) {
+        printf("Launching: %ls\n", cmdline);
+    }
+
     ok = CreateProcessW(NULL, cmdline, NULL, NULL, TRUE, 0, NULL, NULL, &si, &child_process_info);
     if (!ok) {
         // Failed to create process. See if we can find out why.
